### PR TITLE
update model functions

### PR DESCRIPTION
**Pull request by double-y**
### Description
- [x] remove resetting grouping logic from build_lm and build_glm
- [x] prediction function uses source.data column, not given source data frame
- [x] prediction function should have an argument to clean column names of result
- [x] add do_cor as function for ROC curve
- [x] add build_lr for logistic regression
- [x] add evaluation functions for regression and binary classification
- [x] https://github.com/exploratory-io/tam/issues/4279
- [x] fix https://github.com/exploratory-io/tam/issues/4280 (grouping error message handling)

### Checklist

Make sure you have performed following items before submitting this pull request.
If not, please describe the reason.  

- [ ] Add test cases for this fix/enhancement
- [ ] Pass devtools::check()
- [ ] Pass devtools::test()
- [ ] Test installing from github
- [ ] Tested with Exploratory

### CODE REVIEW

File: R/build_lm.R [hunk 1]
```diff
@@ -67,40 +63,35 @@ build_lm <- function(data, ..., keep.source = TRUE, augment = FALSE, group_cols
...
           eval(parse(text = paste0("stats::lm(data = data, ", arg_char, ")")))
         })) %>%
-        dplyr::select(-data) %>%
         dplyr::rowwise()
       class(ret[[source_col]]) <- c("list", ".source.data")
       ret
     } else {
       ret <- data %>%
-        tidyr::nest() %>%
+        tidyr::nest(.key = "source.data") %>%
```
Comment (hideaki) [inline R/build_lm.R]: line 83 to 93 looks same as line 66 to 77. Can we avoid code duplication here?

File: R/model_eval.R [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+    val <- arranged[[actual_val_col]][!is.na(arranged[[actual_val_col]])]
+    if(is.factor(val)){
+      # Need to subtract 1 because the first level in factor is regarded as 1
+      # though it should be FALSE.
+      val <- as.logical(as.integer(val) - 1)
+    } else {
+      val <- as.logical(val)
+    }
+
+    if (all(val)){
```
Comment (hideaki) [inline R/model_eval.R]: Please add some more comment below this point of this function.

File: R/build_lm.R (hunk 1 above)
Comment (double-y) [inline R/build_lm.R]: Yes, it's code duplication. Removed it

File: R/model_eval.R (hunk 2 above)
Comment (double-y) [inline R/model_eval.R]: Okay, will add comments

**Pull request merged**